src/core/Akka/Actor/ActorCell.DefaultMessages.cs

```diff
@@ -376,25 +441,27 @@ public void Resume(Exception causedByFailure)
         /// <summary>
         ///     Async stop this actor
         /// </summary>
+        /// <remarks>➡➡➡ NEVER SEND THE SAME SYSTEM MESSAGE OBJECT TO TWO ACTORS ⬅⬅⬅</remarks>
         public void Stop()
         {
-            SendSystemMessage(Dispatch.SysMsg.Terminate.Instance);
+            SendSystemMessage(new Dispatch.SysMsg.Terminate());
         }
 
         /// <summary>
         ///     Suspends this instance.
         /// </summary>
+        /// <remarks>➡➡➡ NEVER SEND THE SAME SYSTEM MESSAGE OBJECT TO TWO ACTORS ⬅⬅⬅</remarks>
         public void Suspend()
         {
-            SendSystemMessage(Dispatch.SysMsg.Suspend.Instance);
+            SendSystemMessage(new Dispatch.SysMsg.Suspend());
         }
 
+        /// <remarks>➡➡➡ NEVER SEND THE SAME SYSTEM MESSAGE OBJECT TO TWO ACTORS ⬅⬅⬅</remarks>
         public void SendSystemMessage(ISystemMessage systemMessage)
         {
             try
             {
-                // TODO: use a dispatcher or the mailbox to post a message directly
-                _mailbox.Post(Self, new Envelope() { Message = systemMessage, Sender = ActorRefs.NoSender});
+                Dispatcher.SystemDispatch(this, (SystemMessage)systemMessage);
```

Comment:
`SystemDispatch` signals to the mailbox that it's OK to push a new system message into the actor, but not to touch any user messages.